actions/actions.py

```diff
@@ -23,29 +22,122 @@
 
 from reactive.easyrsa import (
     easyrsa_directory,
+    create_client_certificate,
+    create_server_certificate,
 )
 
 # Import charm layers and start reactive
 layer.import_layer_libs()
 hookenv._run_atstart()
 
 PKI_BACKUP = '/home/ubuntu/easyrsa_backup'
-
-
-def backup() -> None:
+# Minimal required contents of the backup tarball
+TAR_STRUCTURE = {'pki',
+                 'pki/ca.crt',
+                 'pki/issued',
+                 'pki/issued/client.crt',
+                 'pki/private',
+                 'pki/private/ca.key',
+                 'pki/private/client.key',
+                 'pki/serial',
+                 }
+
+
+def _ensure_backup_dir_exists():
+    """Ensure that backup directory exists with proper ownership"""
     uid = pwd.getpwnam("ubuntu").pw_uid
     gid = grp.getgrnam("ubuntu").gr_gid
     try:
         os.mkdir(PKI_BACKUP, mode=0o700)
     except FileExistsError:
         pass
-
     os.chown(PKI_BACKUP, uid, gid)
+
     if not os.path.isdir(PKI_BACKUP):
         raise RuntimeError('Backup destination is not a directory.')
 
+
+def _verify_backup(tarball):
+    """
+    Verify that backup archive contains expected files
+
+    :param tarball: Tarfile object containing easyrsa backup
+    :return: None
```

Comment:
:return: option not needed, since the function does not return anything. However, `:raises [ErrorType]: [ErrorDescription]` may be documentes, as the function may raise `RuntimeError`.

Reply:
`:return` removed.
I don't think we need to go into so much detail with `:raises`